### DISCUSSION TITLE
Dear Internet/Coding Gods: Demo Videos/Video Quality

### DISCUSSION THREAD

**Discussion by dansun90** [thygate/stable-diffusion-webui-depthmap-script#192] [Q&A]
Does anyone know if there is a way to output the demo videos/videos generated to a higher resolution other than 720p? to say 1080p?

**Comment by thygate** [comment]
Video resolution is now dependent on the crop-borders field, I recently changed this. This was also broken in the original 3dphoto repo.

**Reply by dansun90** [reply]
thanks for getting back to me! In the Usage section it only has about one sentence related to the crop-borders ("The border can be cropped on four sides...") field. Could you please educate me on how to set the fields to make video 1080p? This is what I'm seeing as the default in SD --> Crop: top, left, bottom, right 
                                       0.03, 0.03, 0.05, 0.03

**Reply by thygate** [reply]
An image with dimensions 1920x1080 and crop borders all zero should give you a 1080p video output. 
Having for instance only top or bottom border 0.5, then the output will be 720p ... etc ..

**Reply by dansun90** [reply]
> An image with dimensions 1920x1080 and crop borders all zero should give you a 1080p video output. 
> 
> Having for instance only top or bottom border 0.5, then the output will be 720p ... etc ..

Thank you! It worked, also playing around/testing. Awesome extension, really appreciate it.

**Reply by thygate** [reply]
I'm sorry I was a bit fast in replying, with crop borders you basically set the crop proportionally to width and height,

so above example, top or bottom at 0.5 would result in 1920x540 video..

**Reply by dansun90** [reply]
no worries! btw, when testing/playing with it, I was looking at the documentation for moviepy and 3d in paint and it seems .avi file format is possible but in the UI for Generate Video tab, limited to only .mp4 and .webm file format. Is there a way to output .avi file? I changed the codec to use PNG instead of libx264 for higher quality video and it was recommended to use .avi file format. 

filename
      Any filename like 'video.mp4' etc. but if you want to avoid
      complications it is recommended to use the generic extension
      '.avi' for all your videos.

codec ``png`` (use file extension ``.avi``) will produce a video
          of perfect quality, of smaller size than with ``rawvideo``.

I tried changing "mp4" for fnExt to 'avi' but still output .mp4 format.

def output_3d_photo(verts, colors, faces, Height, Width, hFov, vFov, tgt_poses, video_traj_types, ref_pose,
                    output_dir, ref_image, int_mtx, config, image, videos_poses, video_basename, original_H=None, original_W=None,
                    border=None, depth=None, normal_canvas=None, all_canvas=None, mean_loc_depth=None, dolly=False, fnExt='avi'):
